.github/workflows/dependabot_auto_merge.yml

```diff
@@ -0,0 +1,23 @@
+name: Dependabot auto-merge
+on: pull_request_target
+
+permissions:
+  pull-requests: write
+  contents: write
+
+jobs:
+  dependabot:
+    runs-on: ubuntu-latest
+    if: ${{ github.event.pull_request.user.login == 'dependabot[bot]' }}
+    steps:
+      - name: Dependabot metadata
+        id: metadata
+        uses: dependabot/fetch-metadata@v1
+        with:
+          github-token: "${{ secrets.GITHUB_TOKEN }}"
+      - name: Enable auto-merge for Dependabot PRs
+        if: ${{ steps.metadata.outputs.update-type != 'version-update:semver-major' && steps.metadata.outputs.dependency-type == 'direct:development' }}
```

Comment:
> steps.metadata.outputs.dependency-type == 'direct:development'
From my understanding of [the docs](https://github.com/marketplace/actions/fetch-metadata-from-dependabot-prs#:~:text=The%20type%20of%20dependency%20has%20determined%20this%20PR%20to%20be.%20Possible%20values%20are%3A%20direct%3Aproduction%2C%20direct%3Adevelopment%20and%20indirect.), this should only allow devDependencies